src/wallets/hardware/LedgerClient.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { DeepPartial } from "ts-essentials";
+import { mock } from "jest-mock-extended";
+import Transport from "@ledgerhq/hw-transport";
+import TransportWebHID from "@ledgerhq/hw-transport-webhid";
+
+interface CreateLedgerClientParams {
+  client?: DeepPartial<TransportWebHID>;
+  transport?: DeepPartial<Transport>;
+}
+
+const createLedgerClient = (params: CreateLedgerClientParams = {}) => {
+  const client = mock<TransportWebHID>(params.client);
+  const transport = mock<Transport>(params.transport);
+
+  jest.mock("@ledgerhq/hw-transport-webhid", () => {
+    return {
+      ...client,
+      create: () => transport,
+    };
+  });
+
+  const LedgerClient = require("./LedgerClient").default;
+
+  return {
+    client: new LedgerClient(),
+    transport,
+  };
+};
+
+afterEach(() => {
+  jest.resetModules();
+});
+
+describe("getVersion", () => {
+  it("returns the current version", async () => {
+    const { client, transport } = createLedgerClient({
+      transport: {
+        // TODO: Examine real response so this mock is reliable.
+        send: jest.fn().mockResolvedValue(Buffer.from([1, 2, 3])),
+      },
+    });
+
+    await client.connect();
+    const result = await client.getVersion();
+
+    expect(transport.send).toHaveBeenCalledWith(128, 6, 0, 0);
```

Comment:
We could export the constants from LedgerClient.ts and reference them here so the tests have a bit more meaning than seemingly random numbers